### PR TITLE
LPS-187725: Localize error messages in API validations

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Create new public static class for `InvalidObjectField` in `ObjectEntryValuesException`
- BuildLang in indicating the expcetion messages
- Update tests cases
- Change syntax of a private method name in `DefaultObjectEntryManagerImpl`

**How to test it?**
- Deploy `object-api` and `object-service`
- Deploy `object-rest-api` and `object-rest-impl`
- Deploy `headless-builder-api` and `headless-builder-impl`

**Related JIRA Ticket**
https://liferay.atlassian.net/browse/LPS-187725

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIApplicationRelevantObjectEntryModelListener.java [hunk 1]
```diff
@@ -89,4 +106,13 @@ private void _validate(ObjectEntry objectEntry) {
 	private static final Pattern _baseURLPattern = Pattern.compile(
 		"[a-zA-Z0-9-]{1,255}");
 
+	@Reference
+	private ObjectDefinitionLocalService _objectDefinitionLocalService;
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIApplicationRelevantObjectEntryModelListener.java:110]: Not needed?

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Review by 4lejandrito — APPROVED** [review]

**Comment by sergiojimcos** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward:force

**Pull request closed without merging**